test/ibmq/test_ibmq_factory.py

```diff
@@ -114,6 +118,19 @@ def test_pass_unreachable_proxy(self, qe_token, qe_url):
             ibmq.enable_account(qe_token, qe_url, proxies=proxies)
         self.assertIn('ProxyError', str(context_manager.exception))
 
+    @requires_qe_access
+    def test_enable_specified_provider(self, qe_token, qe_url):
+        """Test enabling an account with a specified provider."""
+        ibmq = IBMQFactory()
+        non_default_provider = get_provider(ibmq, qe_token, qe_url, default=False)
+        with no_file('Qconfig.py'), custom_qiskitrc(), no_envs(CREDENTIAL_ENV_VARS):
+            hgp = HubGroupProject.from_credentials(non_default_provider.credentials)
```

Comment:
Any reason you need to convert it to a `HubGroupProject` instead of just using `non_default_provider.credentials.hub`?

Reply:
I thought it was more convenient, but definitely unnecessary overhead 😅